extensions/sources/filebase/src/main/java/io/smallrye/ext/config/source/filebase/AbstractUrlBasedSource.java

```diff
@@ -0,0 +1,251 @@
+        this.notifyOnChanges = loadNotifyOnChanges();
+        this.pollForChanges = loadPollForChanges();
+        this.pollInterval = loadPollInterval();
+        this.urlInputString = loadUrlPath();
+        this.loadUrls(urlInputString);
+        
+        super.initOrdinal(500); 
+    }
+    
+    
+    @Override
+    protected Map<String, String> getPropertiesIfEnabled() {
+        return this.properties;
+    }
+
+    @Override
+    public String getValue(String key) {
+        // in case we are about to configure ourselves we simply ignore that key
+        if(super.isEnabled() && !key.startsWith(getPrefix())){
+            return this.properties.get(key);
+        }
+        return null;
+    }
+
+    @Override
+    public String getName() {
+        return getClassKeyPrefix() + UNDERSCORE + this.urlInputString;
+    }
+    
+    @Override
+    public void reload(URL url){
+        Map<String, String> before = new HashMap<>(this.properties);
+        initialLoad(url);
+        mergeProperties();
+        Map<String, String> after = new HashMap<>(this.properties);
+        if(notifyOnChanges)ChangeEventNotifier.getInstance().detectChangesAndFire(before, after,getName());
+    }
+    
+    private void initialLoad(URL url){
+        
+        log.log(Level.INFO, "Using [{0}] as {1} URL", new Object[]{url.toString(), getFileExtension()});
+        
+        InputStream inputStream = null;
+        
+        try {
+            inputStream = url.openStream();
+            if (inputStream != null) {
+                this.propertiesMap.put(url, toMap(inputStream));
+            }
+        } catch (IOException e) {
+            log.log(Level.WARNING, "Unable to read URL [{0}] - {1}", new Object[]{url, e.getMessage()});
+        } finally {
+            try {
+                if (inputStream != null)inputStream.close();
+            // no worries, means that the file is already closed
+            } catch (IOException e) {}
+        }
+    }
+    
+    protected String getKeySeparator(){
+        return this.keySeparator;
+    }
+    
+    private void loadUrls(String surl) {
+        String urls[] = surl.split(COMMA);
+        
+        for(String u:urls){
+            if(u!=null && !u.isEmpty()){
+                loadUrl(u.trim());
+            }
+        }
+        mergeProperties();
+    }
+    
+    private void loadUrl(String url) {
+        try {
+            URL u = new URL(url);
+            initialLoad(u);
+            
+            if(shouldPollForChanges()){
+                // Local (file://...)
+                if(isLocalResource(u)){
+                    enableLocalResourceWatching(u);
+                // Web (http://...)
+                }else if(isWebResource(u)){
+                    enableWebResourceWatching(u);
+                // TODO: Add support for other protocols ?     
+                }else{
+                    log.log(Level.WARNING, "Can not detect changes on resource {0}", u.getFile());
```

Comment:
If this class only supports web and file resources, does it make sense to make it appear general-purpose?

Reply:
What do you suggest? It initially only supported local files, then a request came for web. If more requests comes for more protocols we can add it (that was my thought at least)